backend/station_manager/app/src/test/java/com/StationManager/app/domain/trainstation/HallTest.java

```diff
@@ -0,0 +1,698 @@
+                        "Fname",
+                        "Sname",
+                        (short) 2,
+                        new Privilegy("ordinary", 0),
+                        new Point(12, 9)));
+    }
+
+    @Test
+    @DisplayName("Adding client to single ticket office in right")
+    void testAddClientToSingleRightTicketOffice() {
+        ArrayList<Position> entrances = new ArrayList<>();
+        ArrayList<TicketOffice> ticketOffices = new ArrayList<>();
+
+        Hall hall =
+                new Hall(
+                        new Position(new Point(0, 0), new Point(20, 20)), entrances, ticketOffices);
+
+        Position ticketOfficePosition1 = new Position(new Point(10, 8), new Point(11, 10));
+        TicketOffice ticketOffice1 = new TicketOffice(ticketOfficePosition1, Direction.Right);
+
+        hall.addTicketOffice(ticketOffice1);
+        ticketOffice1.setClosed(false);
+
+        Client client1 =
+                new Client(
+                        2,
+                        "Fname",
+                        "Sname",
+                        (short) 2,
+                        new Privilegy("ordinary", 0),
+                        new Point(3, 4));
+
+        hall.addClient(client1);
+
+        assertEquals(
+                ticketOffice1.getQueue().getClients().get(0),
+                new Client(
+                        2,
+                        "Fname",
+                        "Sname",
+                        (short) 2,
+                        new Privilegy("ordinary", 0),
+                        new Point(9, 9)));
+    }
+
+    @Test
+    @DisplayName(
+            "Adding clients to multiple ticket offices. Should be added to the closest with the"
+                    + " least number of clients")
+    void testAddClientsToMultipleTicketsOffices() {
+        ArrayList<Position> entrances = new ArrayList<>();
+        ArrayList<TicketOffice> ticketOffices = new ArrayList<>();
+
+        Hall hall =
+                new Hall(
+                        new Position(new Point(0, 0), new Point(20, 20)), entrances, ticketOffices);
+
+        Position ticketOfficePosition1 = new Position(new Point(9, 0), new Point(11, 1));
+        Position ticketOfficePosition2 = new Position(new Point(9, 19), new Point(11, 20));
+        Position ticketOfficePosition3 = new Position(new Point(19, 8), new Point(20, 10));
+
+        TicketOffice ticketOffice1 = new TicketOffice(ticketOfficePosition1, Direction.Up);
+        TicketOffice ticketOffice2 = new TicketOffice(ticketOfficePosition2, Direction.Down);
+        TicketOffice ticketOffice3 = new TicketOffice(ticketOfficePosition3, Direction.Right);
+
+        ticketOffice1.setClosed(false);
+        ticketOffice2.setClosed(false);
+        ticketOffice3.setClosed(false);
+
+        hall.addTicketOffice(ticketOffice1);
+        hall.addTicketOffice(ticketOffice2);
+        hall.addTicketOffice(ticketOffice3);
+
+        Client clientRight1 =
+                new Client(
```

Comment:
client creation consumers a lot of space, you can define a method to ease testing where you would just pass an id, and everything else would be dummy values, so you would just do `getClient({id})`